lib/physics/box.go

```diff
@@ -79,7 +91,12 @@ func (b *Box) Contains(position coords.World) bool {
 
 func (b *Box) inSolid(world mapgen.BlockSource) bool {
 	blockCollide := func(x, y, z int) bool {
-		return world.Block(coords.Block{x, y, z}).Solid()
+		blockCoord := coords.Block{x, y, z}
+		if world.Block(blockCoord).Solid() {
+			blockBox := NewBox(blockCoord.Center().Vec3(), vmath.Vec3{0.5, 0.5, 0.5})
+			return b.Collides(blockBox)
+		}
+		return false
```

Comment:
Don't optimize until you need to. Is seems plenty fast enough while playing.
If speed in our logic loop gets to be a problem (this is, our server cannot tick at 60 fps) then we can investigate.
Kind of like how that whole loop fiasco with generating chunks.